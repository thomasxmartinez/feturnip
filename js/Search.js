import React from 'react'
import ShowCard from './ShowCard'
import preload from '../public/data.json'

const Search = React.createClass({
  render () {
    return (
      <div className='search'>
        <div>
          {preload.shows.map((show) => {
            return (
              <ShowCard key={show.imdbID} show={show} />
            )
          })}
        </div>
      </div>
    )
  }
})

export default Search
