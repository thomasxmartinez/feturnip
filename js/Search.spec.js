/* eslint-env node, jest */
import React from 'react'
import { Provider } from 'react-redux'
import Search, { Unwrapped as UnwrappedSearch} from './Search'
import ShowCard from './ShowCard'
import store from './store'
import { setSearchTerm } from './actionCreators'
import { shallow, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import preload from '../public/data.json'

test('Search snapshot', () => {
  const component = shallow(<UnwrappedSearch shows={preload.shows} searchTerm='' />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('Search should render all shows initially', () => {
  const component = shallow(<UnwrappedSearch shows={preload.shows} searchTerm='' />)
  expect(preload.shows.length).toEqual(component.find(ShowCard).length)
})

test('Search should render correct amount of shows based on search', () => {
  const searchWord = 'house'
  store.dispatch(setSearchTerm(searchWord))
  const component = render(<Provider store={store}><Search shows={preload.shows} /></Provider>)
  const showCount = preload.shows.filter((show) => `${show.title.toUpperCase()} ${show.description.toUpperCase()}`.includes(searchWord.toUpperCase())).length
  expect(showCount).toEqual(component.find('.show-card').length)
})
