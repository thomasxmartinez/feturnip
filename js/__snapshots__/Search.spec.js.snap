// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot test 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTU1Mzk2ODEzN15BMl5BanBnXkFtZTgwNDQwMjAxMTI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Girls
        </h3>
        <h4>
          (
          2012–
          )
        </h4>
        <p>
          A comedy about the experiences of a group of girls in their early 20s.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMjEwOTcxODc2Ml5BMl5BanBnXkFtZTgwMjMyMDk2MTI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Game of Thrones
        </h3>
        <h4>
          (
          2011–
          )
        </h4>
        <p>
          Nine noble families fight for control over the mythical lands of Westeros; A forgotten race returns after being dormant for thousands of years.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTU2NjgyNjkyMV5BMl5BanBnXkFtZTgwMTI0MjYyMDI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Insecure
        </h3>
        <h4>
          (
          2016–
          )
        </h4>
        <p>
          Follows the awkward experiences and racy tribulations of a modern-day African-American woman.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BOTA4MTE3MTQwMF5BMl5BanBnXkFtZTgwNzk4MTg4MTI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Silicon Valley
        </h3>
        <h4>
          (
          2014–
          )
        </h4>
        <p>
          Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images.duckduckgo.com/iu/?u=http%3A%2F%2Fpopculture.mearsonlineauctions.com%2FItemImages%2F000054%2F54804a_lg.jpeg&f=1"
      />
      <div>
        <h3>
          Sopranos
        </h3>
        <h4>
          (
          2014
          )
        </h4>
        <p>
          Mob boss deals with issues dealing with business and family
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTYyMzYxMjM3OF5BMl5BanBnXkFtZTcwNjg2OTA3MQ@@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Sex and the City
        </h3>
        <h4>
          (
          2008
          )
        </h4>
        <p>
          A New York writer on sex and love is finally getting married to her Mr. Big. But her three best girlfriends must console her after one of them inadvertently leads Mr. Big to jilt her.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BOTAxOTg2ODM3MF5BMl5BanBnXkFtZTgwMjIyMzgwMzE@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Oz
        </h3>
        <h4>
          (
          1997–2003
          )
        </h4>
        <p>
          A series chronicling the daily activities of an unusual prison facility and its criminal inhabitants.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTI3ODc2ODc0M15BMl5BanBnXkFtZTYwMjgzNjc3._V1_SX300.jpg"
      />
      <div>
        <h3>
          Band of Brothers
        </h3>
        <h4>
          (
          2001
          )
        </h4>
        <p>
          The story of Easy Company of the US Army 101st Airborne division and their mission in WWII Europe from Operation Overlord through V-J Day.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BODUxMDI2Mzc5Nl5BMl5BanBnXkFtZTgwNTQ0OTc2MTI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Veep
        </h3>
        <h4>
          (
          2012–
          )
        </h4>
        <p>
          Former Senator Selina Meyer finds that being Vice President of the United States is nothing like she hoped and everything that everyone ever warned her about.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BOTkwNTc3MjM3MV5BMl5BanBnXkFtZTgwOTU5ODgxOTE@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Ballers
        </h3>
        <h4>
          (
          2015–
          )
        </h4>
        <p>
          A series centered around a group of football players and their families, friends, and handlers.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTU5NjU1MjE0M15BMl5BanBnXkFtZTgwMTQyODgxOTE@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Vice Principals
        </h3>
        <h4>
          (
          2016-
          )
        </h4>
        <p>
          An over protective father is obsessed with becoming a principal and competes against the popular vice principal. However something happens that makes them work together to save the day.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTEyODk5NTc2MjNeQTJeQWpwZ15BbWU4MDQ5NTgwOTkx._V1_SX300.jpg"
      />
      <div>
        <h3>
          Westworld
        </h3>
        <h4>
          (
          2016–
          )
        </h4>
        <p>
          Set at the intersection of the near future and the reimagined past, explore a world in which every human appetite, no matter how noble or depraved, can be indulged without consequence.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMTkyNjIwOTgwM15BMl5BanBnXkFtZTgwNjgwMzA0MTI@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Crashing
        </h3>
        <h4>
          (
          2017–
          )
        </h4>
        <p>
          A New York comic is forced to make a new start for himself after his wife leaves him.
        </p>
      </div>
    </div>
    <div
      className="show-card"
    >
      <img
        src="https://images-na.ssl-images-amazon.com/images/M/MV5BMjNhODE1M2YtMjFmMC00YTliLThhNDktMzI0MjljMzI3NDVjL2ltYWdlXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
      />
      <div>
        <h3>
          Boardwalk Empire
        </h3>
        <h4>
          (
          2010–2014
          )
        </h4>
        <p>
          An Atlantic City politician plays both sides of the law, conspiring with gangsters during the Prohibition era.
        </p>
      </div>
    </div>
  </div>
</div>
`;
